Source/WebCore/platform/graphics/Path.h

```diff
@@ -320,18 +345,17 @@ template<class Decoder> std::optional<Path> Path::decode(Decoder& decoder)
     }
 #endif
 
-    uint64_t numPoints;
-    if (!decoder.decode(numPoints))
-        return std::nullopt;
-
     path.clear();
 
-    for (uint64_t i = 0; i < numPoints; ++i) {
-        PathElement::Type elementType;
+    while (true) {
```

Comment:
(I also wonder if we should have some maximum cap here, so that a compromised web process can't induce an infinite loop on the decoder).

Reply:
Fair point, and response. If that's a problem we wanted to address there are probably other variable length types we'd want to do that for too.